sdk/java/pulumi/src/main/java/com/pulumi/experimental/automation/Workspace.java

```diff
@@ -0,0 +1,569 @@
+     * Workspace.
+     *
+     * @param stackName the stack to operate on
+     * @return the tag map for the specified stack name
+     * @throws AutomationException if there was an issue listing the tags
+     */
+    public abstract Map<String, String> listTags(String stackName) throws AutomationException;
+
+    /**
+     * Returns the value associated with the specified stack name and key, scoped to
+     * the Workspace.
+     *
+     * @param stackName the name of the stack to read config from
+     * @param key       the key to use for the config lookup
+     * @return the value associated with the key
+     * @throws AutomationException if there was an issue reading the config
+     */
+    public ConfigValue getConfig(String stackName, String key) throws AutomationException {
+        return getConfig(stackName, key, false);
+    }
+
+    /**
+     * Returns the value associated with the specified stack name and key, scoped to
+     * the Workspace.
+     *
+     * @param stackName the name of the stack to read config from
+     * @param key       the key to use for the config lookup
+     * @param path      the key contains a path to a property in a map or list to
+     *                  get
+     * @return the value associated with the key
+     * @throws AutomationException if there was an issue reading the config
+     */
+    public abstract ConfigValue getConfig(String stackName, String key, boolean path) throws AutomationException;
+
+    /**
+     * Returns the config map for the specified stack name, scoped to the current
+     * Workspace.
+     *
+     * @param stackName the name of the stack to read config from
+     * @return the config map for the specified stack name
+     * @throws AutomationException if there was an issue listing the config
+     */
+    public abstract Map<String, ConfigValue> getAllConfig(String stackName) throws AutomationException;
+
+    /**
+     * Sets the specified key-value pair in the provided stack's config.
+     *
+     * @param stackName the name of the stack to operate on
+     * @param key       the config key to set
+     * @param value     the config value to set
+     * @throws AutomationException if there was an issue setting the config
+     */
+    public void setConfig(String stackName, String key, ConfigValue value) throws AutomationException {
+        setConfig(stackName, key, value, false);
+    }
+
+    /**
+     * Sets the specified key-value pair in the provided stack's config.
+     *
+     * @param stackName the name of the stack to operate on
+     * @param key       the config key to set
+     * @param value     the config value to set
+     * @param path      the key contains a path to a property in a map or list to
+     *                  set
+     * @throws AutomationException if there was an issue setting the config
+     */
+    public abstract void setConfig(String stackName, String key, ConfigValue value, boolean path)
```

Comment:
Should we proactively expose `--type` (https://github.com/pulumi/pulumi/pull/18287) or at least have a plan for what it would look like if we're not adding it now?